samples/node/cache.js

```diff
@@ -0,0 +1,290 @@
+    contents: [
+      {
+        role: "user",
+        parts: [
+          {
+            fileData: {
+              fileUri: uploadResult.file.uri,
+              mimeType: uploadResult.file.mimeType,
+            },
+          },
+        ],
+      },
+    ],
+  });
+  const cacheName = cacheResult.name; // Save the name for later.
+
+  // Later
+  const getCacheResult = await cacheManager.get(cacheName);
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModelFromCachedContent(getCacheResult);
+  model.generateContent("Please summarize this transcript.");
+  // [END cache_create_from_name]
+  await cacheManager.delete(cacheResult.name);
+}
+
+async function cacheCreateFromChat() {
+  // [START cache_create_from_chat]
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const cacheManager = new GoogleAICacheManager(process.env.API_KEY);
+  const fileManager = new GoogleAIFileManager(process.env.API_KEY);
+
+  const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash-001" });
+  const chat = model.startChat();
+
+  const uploadResult = await fileManager.uploadFile(`${mediaPath}/a11.txt`, {
+    mimeType: "text/plain",
+  });
+
+  let result = await chat.sendMessage([
+    "Hi, could you summarize this transcript?",
+    {
+      fileData: {
+        fileUri: uploadResult.file.uri,
+        mimeType: uploadResult.file.mimeType,
+      },
+    },
+  ]);
+  console.log(`\n\nmodel: ${result.response.text()}`);
+  result = await chat.sendMessage(
+    "Okay, could you tell me more about the trans-lunar injection",
+  );
+  console.log(`\n\nmodel: ${result.response.text()}`);
+
+  const cacheResult = await cacheManager.create({
+    model: "models/gemini-1.5-flash-001",
+    contents: await chat.getHistory(),
+  });
+
+  const newModel = genAI.getGenerativeModelFromCachedContent(cacheResult);
+
+  const newChat = newModel.startChat();
+  result = await newChat.sendMessage(
+    "I didn't understand that last part, could you explain it in simpler language?",
+  );
+  console.log(`\n\nmodel: ${result.response.text()}`);
+  // [END cache_create_from_chat]
+
+  await cacheManager.delete(cacheResult.name);
+}
+
+async function cacheDelete() {
+  // [START cache_delete]
+  const cacheManager = new GoogleAICacheManager(process.env.API_KEY);
+  const fileManager = new GoogleAIFileManager(process.env.API_KEY);
+
+  const uploadResult = await fileManager.uploadFile(`${mediaPath}/a11.txt`, {
+    mimeType: "text/plain",
+  });
+
+  const cacheResult = await cacheManager.create({
+    model: "models/gemini-1.5-flash-001",
+    contents: [
+      {
+        role: "user",
+        parts: [
+          {
+            fileData: {
+              fileUri: uploadResult.file.uri,
+              mimeType: uploadResult.file.mimeType,
+            },
+          },
+        ],
+      },
+    ],
+  });
+  await cacheManager.delete(cacheResult.name);
+  // [END cache_delete]
```

Comment:
The other samples in this file `END` the sample before deleting the cached content. Do we want to do this here, too?

Reply:
No, because this is the delete sample, so we want to show it in the sample.